Web/Controller/Bids.hs

```diff
@@ -157,3 +132,31 @@ validateType item bid = do
                 else bid
 
 
+
+createMailBid :: (?modelContext::ModelContext) => Bid -> IO ()
+createMailBid bid = do
+    item <- fetch (get #itemId bid)
+    itemBids <- fetch (get #bids item)
+    mMailBid <- case getWinningBid itemBids of
+        Nothing -> pure Nothing
+        Just winningBid ->
+            if get #bidType winningBid == Internet || get #price winningBid < 500
+                then do
+                    -- threadDelay (2 * 1000000)
+
+                    mailBid <- newRecord @Bid
+                            |> set #itemId (get #itemId bid)
+                            -- Internet bid type by default.
+                            |> set #bidType AutoMail
+                            |> set #price (get #price winningBid + 10)
+                            |> createRecord
+
+
+                    forkIO $ createMailBid mailBid
```

Comment:
calling this recursively here

Reply:
(I've later removed the forkIO as it's not needed - https://github.com/amitaibu/ihp-bid/commit/64443cc793da40646bc06595524c2adb6c5cb212 )